### ISSUE TITLE
Provide better error output for missing environment variables

### ISSUE THREAD

**Issue by lucperkins** [nushell/nushell#8543]
### Describe the bug

If the environment variable `FOO` isn't set in your environment and you run `$env.FOO` in Nushell, you get an error like this:

```
Error: nu::shell::column_not_found

  × Cannot find column
   ╭─[entry #1:1:1]
 1 │ echo $env.FOO
   ·      ──┬─ ─┬─
   ·        │   ╰── cannot find column 'FOO'
   ·        ╰── value originates here
   ╰────
```

While this output is technically correct, as Nushell sees the environment in terms of columns, I personally think it'd be more helpful to provide a more domain-specific error type and message, such as:

```
Error: nu::shell::env_var_not_set

  × Attempting to access an un-set environment variable
   ╭─[entry #1:1:1]
 1 │ echo $env.FOO
   ·      ──┬─ ─┬─
   ·        │   ╰── environment variable 'FOO' not set
   ·        ╰── value originates here
   ╰────
```

### How to reproduce

Run `$env.{VAR}` for any unset `{VAR}`

### Expected behavior

If I weren't already familiar with Nushell or with a specific bit of Nushell logic, I would expect to get more direct feedback when trying to access an un-set environment variable.

### Configuration

This should be the case with any Nushell config AFAIK.

**Comment by fdncred** [comment]
Agreed. It would be nice to be more verbose with the error message.

**Comment by harshalchaudhari35** [comment]
@rgwood I can take this is up, if still available. I'm new to nushell and rust as a lang but with a bit of steer I can figure it out.

**Comment by rgwood** [comment]
@harshalchaudhari35 This one might be a bit of a stretch for someone new to both Nushell and Rust, but feel free to give it a try.

**Comment by KAAtheWiseGit** [comment]
The problem here is that `follow_cell_path_helper` doesn't know the record's name, and it can't get it from the span, because the global state isn't passed to it.

In theory it could check if the record has `config`/`theme`/`PATH` fields, but this feels more like an ugly hack.
